### PR TITLE
Saving the linked units

### PR DESCRIPTION
**Pull request by matthias-ronge**
This saves the linked processes back to METS.
Please note the explanations in the individual commits. I have tried to better explain why I made the respective changes.

This pull request is based on #2146

### CODE REVIEW

File: Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/DivXmlElementAccess.java [hunk 1]
```diff
@@ -193,20 +194,7 @@ private final void readMetadata(MdSecType mdSec, MdSec mdSecType) {
         super.getMetadata().addAll(metadata);
     }
 
-    /**
```
Comment (Kathrin-Huber) [inline Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/DivXmlElementAccess.java]: why did you remove the javadoc?

File: Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/MetsXmlElementAccess.java [hunk 2]
```diff
@@ -400,7 +400,19 @@ private Mets toMets() {
...
+        List<LinkedStructure> uplinks = workpiece.getUplinks();
+        if (uplinks.isEmpty()) {
+            logical.setDiv(structureRoot);
+        } else {
+            DivType uplinkHolder = new DivType();
+            Mptr uplink = new Mptr();
+            uplink.setHref(uplinks.get(uplinks.size() - 1).getUri().getPath());
+            uplinkHolder.getMptr().add(uplink);
+            uplinkHolder.getDiv().add(structureRoot);
+            logical.setDiv(uplinkHolder);
```
Comment (Kathrin-Huber) [inline Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/MetsXmlElementAccess.java]: setDiv does not sound like  method to put an uplinkHolder...
logical is from type "structMapType" why can I set a div on a type?

**Comment by matthias-ronge** [comment]
Is there still something missing here? Or can you merge the branch now?

File: Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/MetsXmlElementAccess.java (hunk 2 above)
Comment (matthias-ronge) [reply Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/MetsXmlElementAccess.java]: I find that confusing, too. `logical` is the logical `<structMap>`. It is of the “type” structMap (for JAXB, _type_ seems to mean _the name of the XML element_), so JAXB calls the corresponding class `StructMapType`. This does **not refer to** the attribute `TYPE` (=`LOGICAL`) on the XML element.  So I am setting the `<div>` into the `<structMap>` here, as common.

Now for the link: Without link, in `<structMap>` is the `<div>`, which represents the root of the structure node (`structureRoot`). If there is a link up, a meaningless `<div>` is first inserted in `<structMap>`, which only serves as a holder for the `<mptr>` with the link and `structureRoot`. So here `uplinkHolder` is also a `<div>`.

**Comment by matthias-ronge** [comment]
Asking again: Is there still something missing here? Or can you merge the branch now?

**Comment by matthias-ronge** [comment]
This pull request is closed because it is based on a pull request that does not work anymore.
